storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -170,11 +170,10 @@ public static UpdatableStore create(
     childToParentRoot.forEach(treeBuilder::childAndParentRoots);
     final BlockTree blockTree = BlockTree.create(treeBuilder.build(), rootToSlotMap);
     if (blockTree.size() < childToParentRoot.size()) {
-      // This should be an error, but keeping this as a warning now for backwards-compatibility
-      // reasons.  Some existing databases may have unpruned fork blocks, and could become
-      // unusable
-      // if we throw here.  In the future, we should convert this to an error.
-      LOG.warn("Ignoring {} non-canonical blocks", childToParentRoot.size() - blockTree.size());
```

Comment:
Upgraded this to an error because we should no longer be persisting hot blocks that do not descend from the latest finalized block.  This will help us catch regressions more easily with tests.